api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -740,15 +765,57 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
         : String.format("%s.%s", tableAbbreviation, columnName);
   }
 
+  @Override
+  public DbDataset mustGetDbDataset(long workspaceId, long dataSetId) {
```

Comment:
Why move these methods? To me, this kind of validation work feels fine in the controller, though I don't have an objection to having them here instead

Reply:
It was mostly because I wanted to move most of the functionality of generateCodeCells in here and that had a dependency on this function as well as the other validator methods.

Reply:
OK - sounds fine.